lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py

```diff
@@ -289,10 +311,21 @@ def main():
                 # /api/v1/ backwards compat
                 # older versions of tower-cli don't *have* a credential_type
                 # resource
-                params['kind'] = module.params['kind']
+                params['kind'] = module.params.get('kind')
             else:
-                credential_type = credential_type_for_v1_kind(module.params, module)
-                params['credential_type'] = credential_type['id']
+                if module.params.get('credential_type'):
+                    credential_type_res = tower_cli.get_resource('credential_type')
+                    credential_type = credential_type_res.get(name=module.params['credential_type'])
```

Comment:
I'm not a fan of this, because credential type is not unique based on name, and because tower-cli already does a name-based lookup (again, which is wrong) of the related field in the call to the modify method.

But this looks like it's mostly the same as how organization is handled, so I'm trending toward saying this is right for now.